client/src/main/scala/org/http4s/client/Client.scala

```diff
@@ -33,7 +33,7 @@ final case class DisposableResponse[F[_]](response: Response[F], dispose: F[Unit
     val task: F[A] = try f(response)
     catch {
       case e: Throwable =>
-        logger.info(s"Caught exception in otherwise pure expression $e")
+        logger.error(s"Caught exception in otherwise pure expression $e")
```

Comment:
We don't know it was otherwise pure.  And I'm not sure this tells a new FP programmer what they did wrong.

`logger.error(e)("Handled exception in client callback to prevent a connection leak. The callback should always return an `F`. If your callback can fail with an exception you can't handle, call `F.raiseError(exception)`.")